From 121dd9cb020f7e31c9232bdee91c0ee775261c81 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Fri, 27 Jun 2025 09:49:00 -0400
Subject: [PATCH] v4l2codecs: dec: Remove has_videometa member

Now that the code is properly located, this member is not needed
anymore.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/9315>
---
 .../gst-plugins-bad/sys/v4l2codecs/gstv4l2codecav1dec.c   | 8 ++++----
 .../gst-plugins-bad/sys/v4l2codecs/gstv4l2codech264dec.c  | 8 ++++----
 .../gst-plugins-bad/sys/v4l2codecs/gstv4l2codech265dec.c  | 8 ++++----
 .../gst-plugins-bad/sys/v4l2codecs/gstv4l2codecmpeg2dec.c | 8 ++++----
 .../gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp8dec.c   | 8 ++++----
 .../gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp9dec.c   | 8 ++++----
 6 files changed, 24 insertions(+), 24 deletions(-)

diff --git a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecav1dec.c b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecav1dec.c
index 290bfb83cf3..3593f963ebe 100644
--- a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecav1dec.c
+++ b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecav1dec.c
@@ -103,7 +103,6 @@ struct _GstV4l2CodecAV1Dec
   GstV4l2CodecAllocator *src_allocator;
   GstV4l2CodecPool *src_pool;
   gint min_pool_size;
-  gboolean has_videometa;
   gboolean streaming;
   gboolean copy_frames;
   gboolean need_negotiation;
@@ -413,12 +412,13 @@ gst_v4l2_codec_av1_dec_decide_allocation (GstVideoDecoder * decoder,
   GstV4l2CodecAV1Dec *self = GST_V4L2_CODEC_AV1_DEC (decoder);
   GstCaps *caps = NULL;
   guint min = 0, num_bitstream;
+  gboolean has_videometa;
 
   g_clear_object (&self->src_pool);
   g_clear_object (&self->src_allocator);
   g_clear_object (&self->sink_allocator);
 
-  self->has_videometa = gst_query_find_allocation_meta (query,
+  has_videometa = gst_query_find_allocation_meta (query,
       GST_VIDEO_META_API_TYPE, NULL);
 
   gst_query_parse_allocation (query, &caps, NULL);
@@ -427,14 +427,14 @@ gst_v4l2_codec_av1_dec_decide_allocation (GstVideoDecoder * decoder,
     return FALSE;
   }
 
-  if (gst_video_is_dma_drm_caps (caps) && !self->has_videometa) {
+  if (gst_video_is_dma_drm_caps (caps) && !has_videometa) {
     GST_ERROR_OBJECT (self,
         "DMABuf caps negotiated without the mandatory support of VideoMeta");
     return FALSE;
   }
 
   /* Check if we can zero-copy buffers */
-  if (!self->has_videometa) {
+  if (!has_videometa) {
     GstVideoInfo ref_vinfo;
     gint i;
 
diff --git a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codech264dec.c b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codech264dec.c
index 9dd851ab671..feff0c40e12 100644
--- a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codech264dec.c
+++ b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codech264dec.c
@@ -109,7 +109,6 @@ struct _GstV4l2CodecH264Dec
   GstV4l2CodecAllocator *src_allocator;
   GstV4l2CodecPool *src_pool;
   gint min_pool_size;
-  gboolean has_videometa;
   gboolean streaming;
   gboolean interlaced;
   gboolean need_sequence;
@@ -454,6 +453,7 @@ gst_v4l2_codec_h264_dec_decide_allocation (GstVideoDecoder * decoder,
   GstV4l2CodecH264Dec *self = GST_V4L2_CODEC_H264_DEC (decoder);
   GstCaps *caps = NULL;
   guint min = 0, num_bitstream;
+  gboolean has_videometa;
 
   /* If we are streaming here, then it means there is nothing allocation
    * related in the new state and allocation can be ignored */
@@ -464,7 +464,7 @@ gst_v4l2_codec_h264_dec_decide_allocation (GstVideoDecoder * decoder,
   g_clear_object (&self->src_allocator);
   g_clear_object (&self->sink_allocator);
 
-  self->has_videometa = gst_query_find_allocation_meta (query,
+  has_videometa = gst_query_find_allocation_meta (query,
       GST_VIDEO_META_API_TYPE, NULL);
 
   gst_query_parse_allocation (query, &caps, NULL);
@@ -473,14 +473,14 @@ gst_v4l2_codec_h264_dec_decide_allocation (GstVideoDecoder * decoder,
     return FALSE;
   }
 
-  if (gst_video_is_dma_drm_caps (caps) && !self->has_videometa) {
+  if (gst_video_is_dma_drm_caps (caps) && !has_videometa) {
     GST_ERROR_OBJECT (self,
         "DMABuf caps negotiated without the mandatory support of VideoMeta");
     return FALSE;
   }
 
   /* Check if we can zero-copy buffers */
-  if (!self->has_videometa) {
+  if (!has_videometa) {
     GstVideoInfo ref_vinfo;
     gint i;
 
diff --git a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codech265dec.c b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codech265dec.c
index 867bdf2c68b..b2795e6545e 100644
--- a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codech265dec.c
+++ b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codech265dec.c
@@ -109,7 +109,6 @@ struct _GstV4l2CodecH265Dec
   GstV4l2CodecAllocator *src_allocator;
   GstV4l2CodecPool *src_pool;
   gint min_pool_size;
-  gboolean has_videometa;
   gboolean streaming;
   gboolean copy_frames;
   gboolean need_sequence;
@@ -484,6 +483,7 @@ gst_v4l2_codec_h265_dec_decide_allocation (GstVideoDecoder * decoder,
   GstV4l2CodecH265Dec *self = GST_V4L2_CODEC_H265_DEC (decoder);
   GstCaps *caps = NULL;
   guint min = 0, num_bitstream;
+  gboolean has_videometa;
 
   if (self->streaming)
     goto no_internal_changes;
@@ -492,7 +492,7 @@ gst_v4l2_codec_h265_dec_decide_allocation (GstVideoDecoder * decoder,
   g_clear_object (&self->src_allocator);
   g_clear_object (&self->sink_allocator);
 
-  self->has_videometa = gst_query_find_allocation_meta (query,
+  has_videometa = gst_query_find_allocation_meta (query,
       GST_VIDEO_META_API_TYPE, NULL);
 
   gst_query_parse_allocation (query, &caps, NULL);
@@ -501,14 +501,14 @@ gst_v4l2_codec_h265_dec_decide_allocation (GstVideoDecoder * decoder,
     return FALSE;
   }
 
-  if (gst_video_is_dma_drm_caps (caps) && !self->has_videometa) {
+  if (gst_video_is_dma_drm_caps (caps) && !has_videometa) {
     GST_ERROR_OBJECT (self,
         "DMABuf caps negotiated without the mandatory support of VideoMeta");
     return FALSE;
   }
 
   /* Check if we can zero-copy buffers */
-  if (!self->has_videometa) {
+  if (!has_videometa) {
     GstVideoInfo ref_vinfo;
     gint i;
 
diff --git a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecmpeg2dec.c b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecmpeg2dec.c
index b6585228a86..322536f28a0 100644
--- a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecmpeg2dec.c
+++ b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecmpeg2dec.c
@@ -119,7 +119,6 @@ struct _GstV4l2CodecMpeg2Dec
   GstV4l2CodecAllocator *src_allocator;
   GstV4l2CodecPool *src_pool;
   gint min_pool_size;
-  gboolean has_videometa;
   gboolean streaming;
 
   GstMemory *bitstream;
@@ -376,6 +375,7 @@ gst_v4l2_codec_mpeg2_dec_decide_allocation (GstVideoDecoder * decoder,
   GstV4l2CodecMpeg2Dec *self = GST_V4L2_CODEC_MPEG2_DEC (decoder);
   GstCaps *caps = NULL;
   guint min = 0, num_bitstream;
+  gboolean has_videometa;
 
   /* If we are streaming here, then it means there is nothing allocation
    * related in the new state and allocation can be ignored */
@@ -386,7 +386,7 @@ gst_v4l2_codec_mpeg2_dec_decide_allocation (GstVideoDecoder * decoder,
   g_clear_object (&self->src_allocator);
   g_clear_object (&self->sink_allocator);
 
-  self->has_videometa = gst_query_find_allocation_meta (query,
+  has_videometa = gst_query_find_allocation_meta (query,
       GST_VIDEO_META_API_TYPE, NULL);
 
   gst_query_parse_allocation (query, &caps, NULL);
@@ -395,14 +395,14 @@ gst_v4l2_codec_mpeg2_dec_decide_allocation (GstVideoDecoder * decoder,
     return FALSE;
   }
 
-  if (gst_video_is_dma_drm_caps (caps) && !self->has_videometa) {
+  if (gst_video_is_dma_drm_caps (caps) && !has_videometa) {
     GST_ERROR_OBJECT (self,
         "DMABuf caps negotiated without the mandatory support of VideoMeta");
     return FALSE;
   }
 
   /* Check if we can zero-copy buffers */
-  if (!self->has_videometa) {
+  if (!has_videometa) {
     GstVideoInfo ref_vinfo;
     gint i;
 
diff --git a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp8dec.c b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp8dec.c
index 433c1d2bf42..e483caae63a 100644
--- a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp8dec.c
+++ b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp8dec.c
@@ -123,7 +123,6 @@ struct _GstV4l2CodecVp8Dec
   GstV4l2CodecAllocator *src_allocator;
   GstV4l2CodecPool *src_pool;
   gint min_pool_size;
-  gboolean has_videometa;
   gboolean streaming;
   gboolean copy_frames;
 
@@ -337,6 +336,7 @@ gst_v4l2_codec_vp8_dec_decide_allocation (GstVideoDecoder * decoder,
   GstCaps *caps = NULL;
   guint min = 0;
   guint num_bitstream;
+  gboolean has_videometa;
 
   if (self->streaming)
     goto no_internal_changes;
@@ -345,7 +345,7 @@ gst_v4l2_codec_vp8_dec_decide_allocation (GstVideoDecoder * decoder,
   g_clear_object (&self->src_allocator);
   g_clear_object (&self->sink_allocator);
 
-  self->has_videometa = gst_query_find_allocation_meta (query,
+  has_videometa = gst_query_find_allocation_meta (query,
       GST_VIDEO_META_API_TYPE, NULL);
 
   gst_query_parse_allocation (query, &caps, NULL);
@@ -354,14 +354,14 @@ gst_v4l2_codec_vp8_dec_decide_allocation (GstVideoDecoder * decoder,
     return FALSE;
   }
 
-  if (gst_video_is_dma_drm_caps (caps) && !self->has_videometa) {
+  if (gst_video_is_dma_drm_caps (caps) && !has_videometa) {
     GST_ERROR_OBJECT (self,
         "DMABuf caps negotiated without the mandatory support of VideoMeta");
     return FALSE;
   }
 
   /* Check if we can zero-copy buffers */
-  if (!self->has_videometa) {
+  if (!has_videometa) {
     GstVideoInfo ref_vinfo;
     gint i;
 
diff --git a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp9dec.c b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp9dec.c
index bec044715ac..31f4675aa47 100644
--- a/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp9dec.c
+++ b/subprojects/gst-plugins-bad/sys/v4l2codecs/gstv4l2codecvp9dec.c
@@ -123,7 +123,6 @@ struct _GstV4l2CodecVp9Dec
   GstV4l2CodecAllocator *sink_allocator;
   GstV4l2CodecAllocator *src_allocator;
   GstV4l2CodecPool *src_pool;
-  gboolean has_videometa;
   gboolean streaming;
   gboolean copy_frames;
   gboolean need_negotiation;
@@ -610,12 +609,13 @@ gst_v4l2_codec_vp9_dec_decide_allocation (GstVideoDecoder * decoder,
   GstCaps *caps = NULL;
   guint min = 0;
   guint num_bitstream;
+  gboolean has_videometa;
 
   g_clear_object (&self->src_pool);
   g_clear_object (&self->src_allocator);
   g_clear_object (&self->sink_allocator);
 
-  self->has_videometa = gst_query_find_allocation_meta (query,
+  has_videometa = gst_query_find_allocation_meta (query,
       GST_VIDEO_META_API_TYPE, NULL);
 
   gst_query_parse_allocation (query, &caps, NULL);
@@ -624,14 +624,14 @@ gst_v4l2_codec_vp9_dec_decide_allocation (GstVideoDecoder * decoder,
     return FALSE;
   }
 
-  if (gst_video_is_dma_drm_caps (caps) && !self->has_videometa) {
+  if (gst_video_is_dma_drm_caps (caps) && !has_videometa) {
     GST_ERROR_OBJECT (self,
         "DMABuf caps negotiated without the mandatory support of VideoMeta");
     return FALSE;
   }
 
   /* Check if we can zero-copy buffers */
-  if (!self->has_videometa) {
+  if (!has_videometa) {
     GstVideoInfo ref_vinfo;
     gint i;
 
-- 
GitLab

